solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java

```diff
@@ -280,52 +243,10 @@ public float score() throws IOException {
 
       @Override
       public float getMaxScore(int upTo) throws IOException {
-        return Float.POSITIVE_INFINITY;
-      }
-    }
-
-    /**
-     * An iterator that allows to iterate only on the documents for which a feature has
-     * a value.
-     **/
-    public class SolrFeatureScorerIterator extends DocIdSetIterator {
-
-      final private DocIdSetIterator filterIterator;
-      final private DocIdSetIterator scorerFilter;
-
-      SolrFeatureScorerIterator(DocIdSetIterator filterIterator,
-          DocIdSetIterator scorerFilter) {
-        this.filterIterator = filterIterator;
-        this.scorerFilter = scorerFilter;
-      }
-
-      @Override
-      public int docID() {
-        return filterIterator.docID();
-      }
-
-      @Override
-      public int nextDoc() throws IOException {
-        int docID = filterIterator.nextDoc();
-        scorerFilter.advance(docID);
-        return docID;
-      }
-
-      @Override
-      public int advance(int target) throws IOException {
-        // We use iterator to catch the scorer up since
-        // that checks if the target id is in the query + all the filters
-        int docID = filterIterator.advance(target);
-        scorerFilter.advance(docID);
-        return docID;
-      }
-
-      @Override
-      public long cost() {
-        return filterIterator.cost() + scorerFilter.cost();
+        return solrScorer.getMaxScore(upTo);
       }
 
+      // TODO delegate more methods?
```

Comment:
Good question. I'll take a look to see if we could have some reflective unit test coverage either way (similar to existing coverage elsewhere) to ensure delegation for any  methods that might be added in future.

Reply:
Looks like there could be scope for factoring out a `FilterFeatureScorer` class, I've opened https://issues.apache.org/jira/browse/SOLR-14378 re: that.

Reply:
In case it helps, I've branched off your branch here, merged latest master into it and added small [commit](https://github.com/cpoerschke/lucene-solr/commit/56af94b59e0377c4a1279ed5569ba4cc8cea0ee2) for `SolrFeature` here extending `FilterFeatureScorer`.